From 3f306f7908b2f4b50769491d88df20a6632cd2e1 Mon Sep 17 00:00:00 2001
From: Dhaval Shah <dhaval.shah@xilinx.com>
Date: Mon, 13 Jan 2020 00:20:45 -0800
Subject: [PATCH 0807/1851] soc: xilinx: vcu: Divisor calculation for MCU clock
 is updated.

commit 3dfbbefd8cd2e1484d3564820de269b7cd23c97e from
https://github.com/Xilinx/linux-xlnx.git

Divisor of the mcu clock is calculated in such a way that
mcu clock derived from the calculation is greater than
or equal to provided values from the logicoreIP.

Signed-off-by: Dhaval Shah <dhaval.shah@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/soc/xilinx/xlnx_vcu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/xilinx/xlnx_vcu.c b/drivers/soc/xilinx/xlnx_vcu.c
index 98467ac33c1a..d7808a09a11c 100644
--- a/drivers/soc/xilinx/xlnx_vcu.c
+++ b/drivers/soc/xilinx/xlnx_vcu.c
@@ -417,7 +417,7 @@ static int xvcu_set_vcu_pll_info(struct xvcu_device *xvcu)
 				found = cfg;
 				divisor_mcu = pll_clk / mcuclk;
 				mod = pll_clk % mcuclk;
-				if (mcuclk - mod < LIMIT)
+				if (mod != 0)
 					divisor_mcu++;
 				break;
 			}
-- 
2.31.1

